app/routes.js

```diff
@@ -1,43 +1,119 @@
-const express = require('express');
+const express = require("express");
 
-module.exports = (controller) => {
+module.exports = controller => {
     const createApp = () => {
         return express();
     };
 
-    const setup = (app) => {
+    const setup = app => {
         app.use(express.json());
-        app.get('/', (req, res) => {
-            res.send("Welcome to Instagram! <br> Check /api/v1/users/[userId] to view users by ID.");
+
+
+        app.get("/", (req, res) => {
+            res.send(
+                "Welcome to Instagram! <br> Check /api/v1/users/[userId] to view users by ID."
+            );
         });
 
-        app.post('/api/v1/users/', (req, res) => {
-            return controller.createUser(req.body)
+
+        app.get("/api/v1/users/", (req, res) => {
+            return controller
+                .findAllUsers(req)
                 .then(data => {
                     res.send(data);
-                }).catch(err => {
+                })
+                .catch(err => {
+                    res.status(500).send({
+                        message: err.message || "Unable to rerieve users"
+                    });
+                });
+        });
+
+
+        app.post("/api/v1/users/", (req, res) => {
+            return controller
+                .createUser(req.body)
```

Comment:
again about readability probably is better 
 
   let newUserId = reuquest.body.userId 
  ...
  .createUser(newUserId)